app/controllers/api_controller.rb

```diff
@@ -226,4 +227,10 @@ def extend_versioned_controller_actions
       extend "#{version}::#{actions_module}".constantize
     end
   end
+
+  def set_active_storage_host
+    # This is required to access S3 URLs with #service_url, which returns a url that expires in 5 Minutes
+    # In contrast, url_for returns permanent URL, which for privacy/security reasons, we do not want to use.
+    ActiveStorage::Current.host = request.base_url
```

Comment:
Can't we get this from the environment? I could spoof my host and have it change this url to whatever I want? I'm sure that's fine and won't ever be a problem but maybe it's easier to just set a static host, here?

Reply:
We have an environment variable with the full publically-routable address but it is set under CORS_ALLOWED_ORIGINS. I think we may need to have PUBLIC_HOST as well as HOST environment variable set.

Reply:
yes, we could, 
`request.base_url` does already the job,
is there any but what is the benefit of using an Env var?